components/compliance-service/reporting/relaxting/reports.go

```diff
@@ -530,6 +530,212 @@ func (backend *ES2Backend) GetReport(esIndex string, reportId string,
 	return report, errorutils.ProcessNotFound(nil, reportId)
 }
 
+func (backend *ES2Backend) GetControlListItems(from int32, perPage int32, filters map[string][]string,
+	sort string, asc bool) (*reportingapi.ControlItems, error) {
+
+	contListItems := make([]*reportingapi.ControlItem, 0)
+	myName := "GetControlListItems"
+
+	client, err := backend.ES2Client()
+	if err != nil {
+		logrus.Errorf("Cannot connect to ElasticSearch: %s", err)
+		return nil, err
+	}
+
+	// we need to use start_time or else we won't see all of the controls that were used outside of end_time day
+	// we use start_time for the same reason we need it for trends
+	esIndex, err := GetEsIndex(filters, false, true)
+	if err != nil {
+		return nil, errors.Wrap(err, myName)
+	}
+
+	//here, we set latestOnly to true.  We may need to set it to false if we want to search non lastest reports
+	//for now, we don't search non-latest reports so don't do it.. it's slower for obvious reasons.
+	filtQuery := backend.getFiltersQuery(filters, true)
+
+	searchSource := elastic.NewSearchSource().
+		Query(filtQuery).
+		FetchSource(false).
+		Size(1)
+
+	controlTermsAgg := elastic.NewTermsAggregation().Field("profiles.controls.id").
+		Size(100).
```

Comment:
Why do we think `100` makes sense here?

Reply:
We don't.. this is WIP that number by completion will probably default to whatever may be retrieved while remaining performant. We will probably need to cap this at some value though, as this API does not support pagination.